### PR TITLE
Dm 2145

### PR DESCRIPTION
**Pull request by bradjohnson92008**
### JIRA issue link
https://agile6.atlassian.net/browse/DM-2145

## Description - what does this code do?
- Code adds additional fields for publishing: tagline, overview_problem, overview_solution, overview_results.
- Refactor Publishing modal so required fields are called out in the correct section.
- Updated "required" attribute for fields required for publishing not be required when practice is un-published, and required if practice is published (accept for the Practice.Name field - this field is always required - regardless if practice is published or not.

**Required Publishing Fields:**  @practice.name, @practice.initiating_facility_type, @practice.date_initiated, @practice.summary, @practice.support_network_email, @practice.diffusion_histories, @practice.tagline, @practice.overview_problem, @practice.overview_solution, and @practice.overview_results

## Testing done - how did you test it/steps on how can another person can test it 
1.  Login as an admin and go to the /admin pages.
2.  Click on practices and click on "New Practice". 
3.  Fill in required fields and create the new practice.
4.  Edit the new practice and go to the Introduction page.
5.  Blank out the Practice name field and click on Save.
    - client side validation should indicate that Name is required.
6.  Re-populate the name field and click save.
    - Practice should save.
7.  Click on the "Publish" button.  
8.  The publishing modal should display and communicate to users which required fields are missing in order to publish.
image
9.  Fill in all required fields for publishing until you are able to successfully publish the practice.
10.  After publishing the practice go back in to the Introduction, overview, and implementation sections and test that required publishing fields are now required.  If you try to blank them out.. and click "Save" the user should receive a popup that indicates that the field is required in order to save.
image

## Link to mock-ups/mock ups (image file if you have it) (if applicable)
N/A

## Acceptance criteria
- Make sure modal on publish includes tagline
- Ensure required fields for publishing are called out in the right sections (Modal goes section by section)
- Add Tagline to publishing requirement 
- Ensure required fields for publishing. are required after practice is published.
- if practice is not published user can save at will - only required field when un-published is Practice.Name.

## Definition of done
- [ ] Unit tests written (if applicable)
- [ ] e2e/accessibility tests written (if applicable)
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [ ] A link has been provided to the originating JIRA issue
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: spec/features/admin/admin_spec.rb [hunk 1]
```diff
@@ -348,22 +348,21 @@ def visit_practice_show
...
-    create_diffusion_history
-
-    visit '/admin/practices'
-    within_table('index_table_practices') do
-      first('.table_actions').click_link('View')
-    end
-
-    expect(page).to have_selector("input[value='Download Adoption Data']")
-  end
+  # it 'should only display a button to download adoptions if the practice has any' do
```
Comment (emyl3) [inline spec/features/admin/admin_spec.rb:351]: we should also uncomment these to ensure they pass :)

**Review by emyl3 — CHANGES_REQUESTED** [review]
Great job, Brad! Just left a couple comments regarding uncommenting tests haha :)

**Review by JoshingYou1 — APPROVED** [review]
Nice work Brad! Other than commenting the unit tests back in as @emyl3 mentioned, the code worked as intended.

File: spec/features/practice_editor/publish_practice_spec.rb [hunk 2]
```diff
@@ -78,34 +82,34 @@ def set_adoption
...
-      # Publish button should be gone if the practice has been published
-      expect(page).to_not have_link('Publish practice')
-
-      visit practice_path(@practice)
-      expect(page).to have_content('practice summary')
-      expect(page).to have_content('October 1970')
-      expect(page).to have_content('VISN-1')
-      expect(page).to have_content('A public practice')
-    end
+    # it 'Should save and publish the practice if all required fields are met' do
```
[the remark this answers is not in the document]
Comment (bradjohnson92008) [reply spec/features/practice_editor/publish_practice_spec.rb]: Fixed this test.. had to change how the initiating facility is initialized.

File: spec/features/admin/admin_spec.rb (hunk 1 above)
Comment (bradjohnson92008) [reply spec/features/admin/admin_spec.rb]: Fixed this test.. test was creating two practices.. and the first one is blank.. but the second one had the content page was expecting.

**Review by emyl3 — APPROVED** [review]

**Pull request merged**